enhancements/machine-api/multiple-disk-support.md

```diff
@@ -0,0 +1,162 @@
+---
+title: multiple-disk-support
+authors:
+  - "@kannon92"
+reviewers:
+  - "tbd"
+approvers:
+  - "tbd"
+api-approvers:
+  - "tbd"
+creation-date: 2024-09-17
+last-updated: 2024-09-17
+tracking-link:
+  - "https://issues.redhat.com/browse/OCPSTRAT-615"
+  - "https://issues.redhat.com/browse/OCPSTRAT-1592"
+see-also:
+  - "https://github.com/openshift/enhancements/pull/1657"
+replaces:
+superseded-by:
+---
+
+# Disk Support in Openshift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenShift is traditionally a single-disk system, meaning the OpenShift installer is designed to install the OpenShift filesystem on one disk. However, new customer use cases have highlighted the need for the ability to configure additional disks during installation.
+
+## Motivation
+
+Custormers request the ability to add disks for day 0 and day 1 operations. Some of the common areas include designed disk for etcd, dedicated disk for swap partitions, container runtime filesystem, and a separate filesystem for container images.
+
+All of these features are possible to support through a combination of machine configs and machine API changes.
+However, the support varies across the use cases and there is a need to define a general API for disk support so we can have a common experience across different methods.
+
+### Workflow Description
+
+### Goals
+
+TBD
```

Comment:
* Define a common interface for infrastructure platforms to implement to use additional disks for a defined set of specific uses
* Implement common behaviour to safely use the above disks when they have been presented by the infrastructure platform